rllib/core/rl_module/tf/tf_rl_module.py

```diff
@@ -8,8 +8,8 @@
 _, tf, _ = try_import_tf()
 
 
-class TfRLModule(RLModule, tf.keras.Model):
```

Comment:
What broke with the ordering?

Reply:
I'm actually not 100% sure yet. Here are my observations thus far:
- TorchRLModule also has nn.Module first, then RLModule (why?, not sure)
- If we put tf.keras.Model last, then we must specify a proper init c'tor for TfRLModule, b/c the c'tor of keras.Model must be called first (otherwise, keras complains).
- However, if we do write our own c'tor ...
```
def __init__(self, config):
    tf.keras.Model.__init__(self)
    RLModule.__init__(self, config)
```
... then - for some reason - the `setup()` method is called twice and we create two(!) actual modules under the hood.
- My thought then was: put tf.keras.Model first, scrap the c'tor altogether (then python would call the super c'tors in the correct order w/o the duplicate `setup()` call), and we should be good.
- It's this weird multiple-inheritance thing in combination with not using `super().__init__` that screws up things here, I think.

I'll try to poke around and find a clean solution for this.

Reply:
Can we then add comments on the specifics of this choice that we made? This will become a wuestion later again.

Reply:
I'll try to clean this up better (make torch and tf the same) and add an explanation. I'm just worried that we already chose a quite complex inheritance structure and this is what's causing the `setup()` to get called twice. Will figure it out ...

Reply:
That's probably the case. But as long as it's documented it should be fine 😀